### ISSUE TITLE
`Console.ReadKey` behaves differently in .NET 6 comparing to .NET 5 when `TERM` is set to `rxvt`

### ISSUE THREAD

**Issue by daxian-dbw** [dotnet/runtime#63387]
### Description

We got a report in PowerShell repo about "arrow keys not functioning properly in PowerShell 7.2 with `rxvt` terminals", see https://github.com/PowerShell/PowerShell/issues/16606.

It turns out `Console.ReadKey` behaves differently in .NET 6 comparing to .NET 5 when `TERM` is set to `rxvt`.
In .NET 5, the `ConsoleKeyInfo` returned for <kbd>LeftArrow</kbd> doesn't has any modifiers, as shown below:

![image](https://user-images.githubusercontent.com/127450/148151835-f362e1aa-9f01-40e1-aba1-9486f3a2a8f1.png)

**However**, in .NET 6, the the `ConsoleKeyInfo` returned for <kbd>LeftArrow</kbd> has `Control` and `Shift` modifiers specified, as shown below:

![image](https://user-images.githubusercontent.com/127450/148152111-c3b7313e-74cc-406b-8940-64f1bc86c4dc.png)

------------------

It's not like `Console.ReadKey` always work properly in .NET 5 with `rxvt` terminals, for example, <kbd>Home</kbd> returns `Escape+H` (it returns `OH` in .NET 6), and <kbd>End</kbd> returns `Escape+F` (it returns `OF` in .NET 6), instead of `Home` and `End` as expected.

But it's been reported that setting `TERM=xterm` would make `Console.ReadKey` work as expected with `rxvt` terminals.

So, are `rxvt` terminals really supported by .NET 5/6? If so, what configurations should be done to make it work properly with `Console.ReadKey`?

### Reproduction Steps

1. Run `export TERM=rxvt`
2. Start PowerShell 7.2.0 or 7.2.1
3. Try using arrow keys to navigate the cursor position

### Expected behavior

<kbd>LeftArrow</kbd> and <kbd>RightArrow</kbd> should move the cursor left and right;
<kbd>LeftArrow</kbd> and <kbd>RightArrow</kbd> should go through history commands.

### Actual behavior

Arrow keys are not working as expected.
<kbd>Home</kbd> and <kbd>End</kbd> prints `OH` and `OF` respectively.

### Regression?

Yes.

### Known Workarounds

Setting `TERM=xterm` would make `Console.ReadKey` work as expected with `rxvt` terminals.

**Comment by B-Art** [comment]
> ### Description
> We got a report in PowerShell repo about "arrow keys not functioning properly in PowerShell 7.2 with `rxvt` terminals", see [PowerShell/PowerShell#16606](https://github.com/PowerShell/PowerShell/issues/16606).
> 
> It turns out `Console.ReadKey` behaves differently in .NET 6 comparing to .NET 5 when `TERM` is set to `rxvt`. In .NET 5, the `ConsoleKeyInfo` returned for LeftArrow doesn't has any modifiers, as shown below:
> 
> ![image](https://user-images.githubusercontent.com/127450/148151835-f362e1aa-9f01-40e1-aba1-9486f3a2a8f1.png)
> 
> **However**, in .NET 6, the the `ConsoleKeyInfo` returned for LeftArrow has `Control` and `Shift` modifiers specified, as shown below:
> 
> ![image](https://user-images.githubusercontent.com/127450/148152111-c3b7313e-74cc-406b-8940-64f1bc86c4dc.png)
> 
> It's not like `Console.ReadKey` always work properly in .NET 5 with `rxvt` terminals, for example, Home returns `Escape+H` (it returns `OH` in .NET 6), and End returns `Escape+F` (it returns `OF` in .NET 6), instead of `Home` and `End` as expected.
> 
> But it's been reported that setting `TERM=xterm` would make `Console.ReadKey` work as expected with `rxvt` terminals.
> 
> So, are `rxvt` terminals really supported by .NET 5/6? If so, what configurations should be done to make it work properly with `Console.ReadKey`?
> 
> ### Reproduction Steps
> 1. Run `export TERM=rxvt`
> 2. Start PowerShell 7.2.0 or 7.2.1
> 3. Try using arrow keys to navigate the cursor position
> 
> ### Expected behavior
> LeftArrow and RightArrow should move the cursor left and right; LeftArrow and RightArrow should go through history commands.
> 
> ### Actual behavior
> Arrow keys are not working as expected. Home and End prints `OH` and `OF` respectively.
> 
> ### Regression?
> Yes.
> 
> ### Known Workarounds
> Setting `TERM=xterm` would make `Console.ReadKey` work as expected with `rxvt` terminals.
> 
> ### Configuration
> _No response_
> 
> ### Other information
> _No response_

In expected behavior:
You mean up arrow ⬆️ and down arrow ⬇️ for going through history.

**Comment by daxian-dbw** [comment]
> You mean up arrow ⬆️ and down arrow ⬇️ for going through history.

Yes, it was reported that arrow keys worked as expected after setting `TERM=xterm`. I have updated the PR description to make this more clear:

> But it's been reported that setting `TERM=xterm` would make `Console.ReadKey` work as expected with regard to arrow keys with `rxvt` terminals -- see https://github.com/PowerShell/PowerShell/issues/16606#issuecomment-997765659.

**Comment by maxim-lobanov** [comment]
Any updates on this issue? It causes issues in PowerShell Core 7.2 because of update to .NET 6.
